src/main/java/com/artipie/asto/google/GoogleStorage.java

```diff
@@ -0,0 +1,121 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto.google;
+
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Remaining;
+import com.artipie.asto.Storage;
+import com.artipie.asto.Transaction;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.StorageOptions;
+import io.reactivex.Flowable;
+import java.io.ByteArrayOutputStream;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Storage that holds data in Google storage.
+ *
+ * @since 0.1
+ */
+@SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+public final class GoogleStorage implements Storage {
+
+    /**
+     * Google storage client.
+     */
+    private final com.google.cloud.storage.Storage client;
+
+    /**
+     * Bucket name.
+     */
+    private final String bucket;
+
+    /**
+     * Ctor.
+     *
+     * @param bucket Bucket name
+     */
+    public GoogleStorage(final String bucket) {
+        this(StorageOptions.getDefaultInstance(), bucket);
+    }
+
+    /**
+     * Ctor.
+     * @param options Options for Google storage client
+     * @param bucket Bucket name
+     */
+    public GoogleStorage(final StorageOptions options, final String bucket) {
+        this.client = options.getService();
+        this.bucket = bucket;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> exists(final Key key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Collection<Key>> list(final Key prefix) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Void> save(final Key key, final Content content) {
+        return CompletableFuture.runAsync(
+            () -> {
+                final ByteArrayOutputStream output = new ByteArrayOutputStream();
+                Flowable.fromPublisher(content).blockingForEach(
+                    buffer -> output.write(new Remaining(buffer).bytes())
+                );
+                this.client.create(
+                    BlobInfo.newBuilder(BlobId.of(this.bucket, key.string())).build(),
+                    output.toByteArray()
+                );
+            });
+    }
+
+    @Override
+    public CompletableFuture<Void> move(final Key source, final Key destination) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Content> value(final Key key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Void> delete(final Key key) {
+        return null;
```

Comment:
inconsistent with other placeholders

Reply:
@Sammers21 it's fixed in this pr: #120. This one is not supposed to be assigned now, as it contains commits from that pr. We need to merge that one first, then I will rebase this one on #120 and only then it can be reviewed. But I told that several times to @g4s8. Hopefully it will be merged soon.